KenticoCloud.Delivery/ContentItemsInRichText/ContentItemsInRichTextProcessor.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using AngleSharp.Dom;
+using AngleSharp.Parser.Html;
+using AngleSharp.Parser.Xml;
+using AngleSharp.Services.Default;
+
+namespace KenticoCloud.Delivery.ContentItemsInRichText
+{
+    /// <summary>
+    /// Processor responsible for parsing richtext elements and resolving content items referenced in them using provided resolvers.
+    /// </summary>
+    public class ContentItemsInRichTextProcessor
+    {
+        private readonly Dictionary<Type, Func<object, string>> _typeResolver;
+
+        private readonly IContentItemsInRichTextResolver<UnretrievedContentItem> _unretrievedContentItemsInRichTextResolver;
+
+        /// <summary>
+        /// Resolver used in case we haven't registered other resolver for processed content type in richtext.
+        /// </summary>
+        public IContentItemsInRichTextResolver<object> DefaultResolver;
+
+        /// <summary>
+        /// Rich text output processor, going through HTML and replacing content items marked as object elements with output of resolvers.
+        /// </summary>
+        /// <param name="defaultResolver">Resolver used in case we haven't registered content type specific resolver.</param>
+        /// <param name="unretrievedContentItemsInRichTextResolver">Resolver whose output is used in case we haven't retrieved value of content item in richtext, 
+        /// this can happen if depth of client request is too low.</param>
+        public ContentItemsInRichTextProcessor(IContentItemsInRichTextResolver<object> defaultResolver, IContentItemsInRichTextResolver<UnretrievedContentItem> unretrievedContentItemsInRichTextResolver)
+        {
+            DefaultResolver = defaultResolver;
+            _typeResolver = new Dictionary<Type, Func<object, string>>();
+            _unretrievedContentItemsInRichTextResolver = unretrievedContentItemsInRichTextResolver;
+        }
+
+        /// <summary>
+        /// Processes richtext output, which is expected to be in HTML format and returns it with resolved content items.
+        /// </summary>
+        /// <param name="value">Richtext output.</param>
+        /// <param name="usedContentItems">Content items used in richtext, used as input for resolution function.</param>
+        /// <returns></returns>
+        public string Process(string value, Dictionary<string, object> usedContentItems)
```

Comment:
This is more of a generic question - how are we going to tackle the situation where the developer wants to resolve content items differently in different parts of their application? Do we want to support that?
ping @kenticomartinh